internal/config/config.go

```diff
@@ -95,6 +96,13 @@ func (s *Config) Load() error {
 		}
 	}
 
+	if v.GetString("username") != "" && v.GetString("password") == "" {
+		password, err := keyring.Get("upctl", v.GetString("username"))
```

Comment:
Just one quick nitty comment based on skimming, not a full review: let's extract the service name to a constant, and reuse in the missing_credentials.go message and tests.